transport-native-epoll/src/main/java/io/netty/channel/epoll/AbstractEpollChannel.java

```diff
@@ -48,12 +49,24 @@
         readFlag = flag;
         flags |= flag;
         this.active = active;
+        fileDescriptor = new FileDescriptor(fd);
     }
 
     protected int fd() {
         return fd;
     }
 
+    /**
+     * Returns the {@link FileDescriptor} that is used by this {@link Channel}.
+     */
+    public FileDescriptor fileDescriptor() {
```

Comment:
Or just remove the old `fd()` method and let the callers do `fd().intValue()` instead.